server/config/config.go

```diff
@@ -433,7 +434,13 @@ func (c *Configurator) GetCacheRedisTarget() string {
 	if redisConfig := c.GetCacheRedisConfig(); redisConfig != nil {
 		return redisConfig.RedisTarget
 	}
-	return c.gc.Cache.RedisTarget
+	if c.gc.Cache.RedisTarget != "" {
+		return c.gc.Cache.RedisTarget
+	}
+	if dcc := c.GetDistributedCacheConfig(); dcc != nil && dcc.RedisTarget != "" {
+		return dcc.RedisTarget
+	}
```

Comment:
Can you separate this out?

The distributed cache uses redis for discovery only. It should be an error to configure the distributed cache without a redis_target of its own in the distributed_cache: block.